docs/content/modeling/modular-models.mdx

```diff
@@ -0,0 +1,189 @@
+
+- A model can grow large and difficult to understand
+- As more teams begin to contribute to a model, the ownership boundaries may not be clear and code review processes might not scale
+
+With modular models, a single model can be split across multiple files in a project. Organized in a way that makes sense for the project or teams, and then ownership for reviews can be expressed using a feature such as [GitHub](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) or [GitLab](https://docs.gitlab.com/ee/user/project/codeowners/) code owners.
+
+## Key Concepts
+
+### `fga.mod`
+
+The `fga.mod` is the project file for modular models. This file is used to track the schema version for the final combined model and the individual files that make up the modular model.
+
+| Property | Description |
+| -------- | -------- | 
+| `schema` | The schema version to be used for the combined model |
+| `contents` | The individual files that make up the modular model |
+
+### Modules
+
+Modules are declared using the `module` keyword in the DSL, and a module can be written across multiple files.
+
+Currently, modules are stored as metadata but are not used by <ProductName format={ProductNameFormat.ShortForm}/>. However, modules will be the basis of future features such as applying authorization to writing and reading/querying tuples.
+
+### Type Extensions
+
+As teams implement features, they might find that core types they are dependent upon might not contain all the realtions they need. However, it might not make sense for these relations to be owned by the owner of that type if they aren't needed across the system.
+
+To solve this, individual types can be extended within other modules to implement the relations needed.
+
+In order to allow this, there are certain requirements for type extension:
+
+- The extended type must exist
+- A single type can only be extended once per file
+- The relations added must not already exist, or be part of another type extension
+
+## Example
+
+In this example we'll look at how an authorization model for a SaaS compny with a issue tracking and wiki software could implement modular models.
+
+### Core
+
+It's likely there will be a core set of types owned by a team that manages the overall identity for the company, this would provide the basics usch as users, organizations and groups that can be used by each respective product area.
+
+```
+module core
+
+type user
+
+type organization
+  relations
+    define member: [user]
+    define admin: [user]
+
+type group
+  relations
+    define member: [user]
+```
+
+### Issue tracking
+
+In the issue tracking software we'd likely separate out the project and issue model into separate files, we'll also extend the `organization` type here so that we can add a relation to track who can create a project.
```

Comment:
```suggestion
In the issue tracking software we'd likely separate out the project and issue related types into separate files, we'll also extend the `organization` type here so that we can add a relation specific to the issue tracking feature, which is the ability to authorize who can create a project.
```